### PR TITLE
fix(casperf): invalid next offset calculation

### PR DESCRIPTION
**Pull request by tiagolobocastro**
Also add easy block size specification, eg: allows "1MiB". Disable cleanup of nexuses when they're not configured. Prevent double cleanup with duplicate handlers.

### CODE REVIEW

File: io-engine/src/bin/casperf.rs [hunk 1]
```diff
@@ -130,12 +130,12 @@ impl Job {
             return;
         }
 
-        let offset = (job.rng.gen::<u64>() % job.io_size) * job.io_blocks;
+        let offset = job.rng.gen_range(0 .. job.io_blocks) * job.io_size;
```
Comment (dsharma-dc) [inline io-engine/src/bin/casperf.rs:133]: how do we ensure this doesn't span beyond device size?

**Review by dsharma-dc — APPROVED** [review]

File: io-engine/src/bdev/nexus/nexus_module.rs [hunk 2]
```diff
@@ -21,13 +21,21 @@ pub(crate) struct NexusModule {}
...
     /// Panics if the Nexus module was not registered.
     pub fn current() -> BdevModule {
         match BdevModule::find_by_name(NEXUS_MODULE_NAME) {
             Ok(m) => m,
             Err(err) => panic!("{}", err),
         }
     }
+    /// Returns Nexus Bdev module instance, if registered.
+    pub fn current_opt() -> Option<BdevModule> {
+        match BdevModule::find_by_name(NEXUS_MODULE_NAME) {
```
Comment (mtzaurus) [inline io-engine/src/bdev/nexus/nexus_module.rs]: IMHO, you can use just '.ok()' method to get the resulting Option:
```
BdevModule::find_by_name(NEXUS_MODULE_NAME).ok()
```

**Review by mtzaurus — APPROVED** [review]

File: io-engine/src/bdev/nexus/nexus_module.rs (hunk 2 above)
Comment (tiagolobocastro) [reply io-engine/src/bdev/nexus/nexus_module.rs]: yes, indeed

File: io-engine/src/bin/casperf.rs (hunk 1 above)
Comment (tiagolobocastro) [reply io-engine/src/bin/casperf.rs:133]: I believe that's ensured by the io_blocks calculation, which is the size of the device in io_size units.

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**